actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -351,8 +351,8 @@ def each_pair(&block)
     # Convert all hashes in values into parameters, then yield each value in
     # the same way as <tt>Hash#each_value</tt>.
     def each_value(&block)
-      @parameters.each_pair do |key, value|
-        yield [convert_hashes_to_parameters(key, value)]
+      @parameters.each_pair do |_, value|
+        yield convert_value_to_parameters(value)
```

Comment:
> `each_value` should yield with "value" of the params instead of "value" as an array.

I'm curious, does this mean that yield with "value" as an array would cause unexpected results?

Reply:
Yep, It is unexpected that passed value to the block of `ActionController::Parameters#each_value` is always an Array. I think the behavior of the method should be similar to `Hash#each_value`. I just added example "before-afrer" to the description of the PR in order to make it clear for rewires - https://github.com/rails/rails/pull/34210#issue-222725782.

Reply:
is there any reason to change `convert_hashes_to_parameters` to `convert_value_to_parameters`?

https://github.com/rails/rails/blob/4de7fdbc37ea002afb68e7c9bf581c34d7cdd566/actionpack/lib/action_controller/metal/strong_parameters.rb#L841-L859